tests/test_references.py

```diff
@@ -205,3 +204,151 @@ def test_forward_references_not_fulfilled__nickname(self, write_row):
+            mock.call("A", {"id": 6, "B_ref": "B(6)"}),
+            mock.call("B", {"id": 6, "A_ref": "A(6)"}),
+        ]
+
+    def test_forward_reference__iterations__with_just_once(self, generated_rows):
+        yaml = """
+            - object: A
+              fields:
+                B_ref:
+                  reference:
+                    B_nick
+            - object: B
+              just_once: True
+              nickname: B_nick
+              fields:
+                A_ref:
+                  reference:
+                    A
+              """
+        self._generate_loop(yaml, 3, 2)
+        assert generated_rows.mock_calls == [
+            mock.call("A", {"id": 1, "B_ref": "B(1)"}),
+            mock.call("B", {"id": 1, "A_ref": "A(1)"}),
+            mock.call("A", {"id": 2, "B_ref": "B(1)"}),
+            mock.call("A", {"id": 3, "B_ref": "B(1)"}),
+            mock.call("A", {"id": 4, "B_ref": "B(1)"}),
+            mock.call("A", {"id": 5, "B_ref": "B(1)"}),
+            mock.call("A", {"id": 6, "B_ref": "B(1)"}),
+        ]
+
+    def test_random_reference_simple(self, generated_rows):
+        yaml = """                  #1
+      - object: A                   #2
+        count: 10                   #4
+      - object: B                   #5
+        count: 2                    #6
+        fields:                     #7
+            A_ref:                  #8
+              random_reference: A   #9
+    """
+        with mock.patch("random.randint") as randint:
+            randint.side_effect = [8, 3]
+            generate(StringIO(yaml))
+        assert generated_rows.row_values(10, "A_ref") == "A(8)"
+        assert generated_rows.row_values(11, "A_ref") == "A(3)"
+
+    def test_random_reference_local(self, generated_rows):
+        yaml = """                  #1
+      - object: A                   #2
+        count: 10                   #4
+      - object: B                   #5
+        fields:                     #7
+            A_ref:                  #8
+              random_reference:     #9
+                tablename: A        #10
+                scope: local        #11
+    """
+        with mock.patch("random.randint") as randint:
+            randint.side_effect = [8, 12]
+            generate(StringIO(yaml), stopping_criteria=StoppingCriteria("B", 2))
+            assert randint.mock_calls == [mock.call(1, 10), mock.call(11, 20)]
+        assert generated_rows.table_values("B", 2, "A_ref") == "A(12)"
+
+    def test_random_reference_global(self, generated_rows):
+        # undocumented experimental feature!
+        yaml = """                  #1
+      - object: A                   #2
+        count: 10                   #4
+      - object: B                   #5
+        count: 2                    #6
+        fields:                     #7
+            A_ref:                  #8
+              random_reference:
+                tablename: A   #9
+                scope: global
```

Comment:
`scope: current-iteration` (default) vs `scope: prior-and-current-iterations` might be clearer naming? (if a little verbose)

Reply:
Renamed in f3b2629 . (It's still a secret feature so I'm not going to workshop the name too much but your suggestion is clear and could always be changed/aliased later)